### ISSUE TITLE
A couple of possible issues with cA code

### ISSUE THREAD

**Issue by gsanroma** [lisa-lab/DeepLearningTutorials#164]
Hi, there might be a couple of issues in the cA code.
First, when computing the loss due to the jacobian (line 211) shouldn't the integer division (//) be replaced by a a true division (/) ?
Second, the result of the previous line of code is a scalar. Then, why taking its mean when computing the cost function in next line of code (218) ?
Thanks,
Gerard.

**Comment by lamblin** [comment]
> First, when computing the loss due to the jacobian (line 211) shouldn't the integer division (//) be replaced by a a true division (/) ?

Yes, this should be a true division. This seems to be an issue that has been introduced when trying to make the code compatible with python 3, but that line should not have been changed, since `J` (and then `sum(J ** 2)`) should be float.

A pull request to fix that would be welcome.

> Second, the result of the previous line of code is a scalar. Then, why taking its mean when computing the cost function in next line of code (218) ?

It is indeed unnecessary, but should not affect the computation speed. That fix would be welcome as well.

Thanks for the report.

**Comment by gsanroma** [comment]
Is there any tutorial on how to create pull requests ?

**Comment by lamblin** [comment]
You can have a look at http://deeplearning.net/software/theano/dev_start_guide.html, in particular http://deeplearning.net/software/theano/dev_start_guide.html#installation-and-configuration and http://deeplearning.net/software/theano/dev_start_guide.html#development-workflow, replacing `Theano/Theano` by `lisa-lab/DeepLearningTutorials`.
